src/utils/decorators.ts

```diff
@@ -48,6 +48,11 @@ export function createClassPropertyDecorator(
 					}
 				}
 			};
+			if (arguments.length < 3) {
+				// Typescript target is ES3, so it won't define property for us
+				Object.defineProperty(target, key, descriptor);
```

Comment:
Ha sorry yeah read this in the wrong other :) Thanks for sorting this out!
I'll run some more tests but I think this will do.

Op ma 20 jun. 2016 om 16:32 schreef Andy Kogut <notifications@github.com>:

> In src/utils/decorators.ts
> <https://github.com/mobxjs/mobx/pull/343#discussion_r67699836>:
>
> > @@ -48,6 +48,11 @@ export function createClassPropertyDecorator(
> >  					}
> >  				}
> >  			};
> > +			if (arguments.length < 3) {
> > +				// Typescript target is ES3, so it won't define property for us
> > +				Object.defineProperty(target, key, descriptor);
>
> I mean some people can forget to specify typescript target, i thought, it
> would be nice if mobx still works. Fixing jsfiddle is also nice.
> And it fixes Reflect.decorate (#333
> <https://github.com/mobxjs/mobx/issues/333>)
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx/pull/343/files/2f55b6b7de579e7aa3c2a5ca2f552f2c827eb0b0#r67699836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABvGhPWGhpfY_7NxHfvaK6O_iyrYXiFPks5qNqSGgaJpZM4I5MzZ>
> .
>